pkg/providers/imagefamily/bootstrap/parser_helpers.go

```diff
@@ -245,3 +258,260 @@ func getHasSearchDomain(csd *nbcontractv1.CustomSearchDomain) bool {
 	}
 	return false
 }
+
+func getCSEHelpersFilepath() string {
+	return cseHelpersScriptFilepath
+}
+
+func getCSEDistroHelpersFilepath() string {
+	return cseHelpersScriptDistroFilepath
+}
+
+func getCSEInstallFilepath() string {
+	return cseInstallScriptFilepath
+}
+
+func getCSEDistroInstallFilepath() string {
+	return cseInstallScriptDistroFilepath
+}
+
+func getCSEConfigFilepath() string {
+	return cseConfigScriptFilepath
+}
+
+func getCustomSearchDomainFilepath() string {
+	return customSearchDomainsCSEScriptFilepath
+}
+
+func getDHCPV6ServiceFilepath() string {
+	return dhcpV6ServiceCSEScriptFilepath
+}
+
+func getDHCPV6ConfigFilepath() string {
+	return dhcpV6ConfigCSEScriptFilepath
+}
+
+// getSysctlContent converts nbcontractv1.SysctlConfig to a string with key=value pairs, with default values.
+//
+//gocyclo:ignore
+func getSysctlContent(s *nbcontractv1.SysctlConfig) string {
+	// This is a partial workaround to this upstream Kubernetes issue:
+	// https://github.com/kubernetes/kubernetes/issues/41916#issuecomment-312428731
+
+	if s == nil {
+		// If the sysctl config is nil, setting it to non-nil so that it can go through the defaulting logic below to get the default values.
+		s = &nbcontractv1.SysctlConfig{}
+	}
+
+	m := make(map[string]int32)
+	m["net.ipv4.tcp_retries2"] = 8
+	m["net.core.message_burst"] = 80
+	m["net.core.message_cost"] = 40
+
+	// Access the variable directly, instead of using the getter, so that it knows whether it's nil or not.
+	// This is based on protobuf3 explicit presence feature.
+	// Other directly access variables in this function implies the same idea.
+	if s.NetCoreSomaxconn == nil {
+		m["net.core.somaxconn"] = 16384
+	} else {
+		// either using getter for NetCoreSomaxconn or direct access is fine because we ensure it's not nil.
+		m["net.core.somaxconn"] = s.GetNetCoreSomaxconn()
+	}
+
+	if s.NetIpv4TcpMaxSynBacklog == nil {
+		m["net.ipv4.tcp_max_syn_backlog"] = 16384
+	} else {
+		m["net.ipv4.tcp_max_syn_backlog"] = s.GetNetIpv4TcpMaxSynBacklog()
+	}
+
+	if s.NetIpv4NeighDefaultGcThresh1 == nil {
+		m["net.ipv4.neigh.default.gc_thresh1"] = 4096
+	} else {
+		m["net.ipv4.neigh.default.gc_thresh1"] = s.GetNetIpv4NeighDefaultGcThresh1()
+	}
+
+	if s.NetIpv4NeighDefaultGcThresh2 == nil {
+		m["net.ipv4.neigh.default.gc_thresh2"] = 8192
+	} else {
+		m["net.ipv4.neigh.default.gc_thresh2"] = s.GetNetIpv4NeighDefaultGcThresh2()
+	}
+
+	if s.NetIpv4NeighDefaultGcThresh3 == nil {
+		m["net.ipv4.neigh.default.gc_thresh3"] = 16384
+	} else {
+		m["net.ipv4.neigh.default.gc_thresh3"] = s.GetNetIpv4NeighDefaultGcThresh3()
+	}
+
+	if s.NetCoreNetdevMaxBacklog != nil {
```

Comment:
is there a reason why you can't directly use the getter's here and all the ones below? the getters are nil safe so i dont think you need the nil check.

Reply:
I left a comment in line 311 too in case later we want to know why.

Reply:
ahh oops i missed that, okay makes sense!